dashboard/app/controllers/script_levels_controller.rb

```diff
@@ -385,13 +385,13 @@ def get_hidden_stage_ids(script_name)
     if !script_sections.empty?
       # if we have sections matching this script id, we consider a stage hidden only if it is hidden in every one
       # of the sections the student belongs to that match this script id
-      all_ids = script_sections.map(&:section_hidden_stages).flatten.map(&:stage_id)
+      all_ids = script_sections.map(&:section_hidden_stages).flatten.pluck(:stage_id)
```

Comment:
Note, `script_sections` is an `Array` due to the [select](https://apidock.com/rails/ActiveRecord/QueryMethods/select) on line 383. This could be made more efficient with a `join` and `where` instead. 

TIL [pluck](http://www.rubydoc.info/github/rails/rails/Enumerable:pluck) works on arrays.
Apparently, this was [added](http://blog.bigbinary.com/2016/02/17/active-support-improvements-in-Rails-5.html) in Rails 5. Before that, this line would have failed, but it's fine now.

Also `.map(&:section_hidden_stages).flatten` can be condensed into `.flat_map(&:section_hidden_stages)`. It's not a perf improvement, but a bit more concise.